pkgs/development/compilers/pforth/default.nix

```diff
@@ -14,14 +15,25 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-vEjFeHSJl+yAtatYJEnu+r9hmOr/kZOgIbSUXR/c8WU=";
   };
 
-  dontConfigure = true;
-
-  preBuild = ''
-    cd platforms/unix
+  # We build the dictionary in a cross-compile compatible way.
+  # For that, we perform steps, that the Makefile would otherwise do.
+  buildPhase = ''
+    runHook preBuild
+    make -C platforms/unix pfdicapp
+    (cd fth/
+      ${stdenv.hostPlatform.emulator buildPackages} ../platforms/unix/pforth -i system.fth
```

Comment:
What this emulator does?

Reply:
In a direct compile scenario ( e.g. on x86_64-linux ) this evaluates to:

```sh
$ nix-instantiate --eval -E "with import <nixpkgs> {}; with stdenv; hostPlatform.emulator buildPackages"
"/nix/store/51sszqz1d9kpx480scb1vllc00kxlx79-bash-5.2-p15/bin/bash -c '\"$@\"' --"
```

In a cross-compile scenario this evaluates to:

```sh
$ nix-instantiate --eval -E "with import <nixpkgs> {}; with pkgsCross.raspberryPi.stdenv; hostPlatform.emulator buildPackages"
"/nix/store/2vpnb8pq7x9fy41fp5v5px4hyw9x3r97-qemu-8.0.2/bin/qemu-arm"
```

This is needed because we need to create this forth dictionary with the machine architecture, that we are compiling to.